### PR TITLE
[Merged by Bors] - feat: Port/data.nat.succ_pred

### PR DESCRIPTION
**Pull request by casavaca**
---

- [x] depends on: #1084

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1084
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1084~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Nat/SuccPred.lean [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Fin.Basic
+import Mathlib.Order.SuccPred.Basic
+
+/-!
+# Successors and predecessors of naturals
+
+In this file, we show that `ℕ` is both an archimedean `succ_order` and an archimedean `pred_order`.
```
Comment (jcommelin) [inline Mathlib/Data/Nat/SuccPred.lean:17]:
```suggestion
In this file, we show that `ℕ` is both an archimedean `succOrder` and an archimedean `predOrder`.
```
Also elsewhere

**Review by jcommelin — COMMENTED** [review]
Please fix some naming issues.

Otherwise, LGTM

bors d+

**Pull request closed without merging**